.github/workflows/calibration.yml

```diff
@@ -0,0 +1,83 @@
+name: Test Lambda Function Locally and Upload Artifacts
+
+on: [pull_request_target]
+    
+jobs:
+  test-and-upload:
+    permissions:
+        pull-requests: write
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+
+    - name: Build Lambda Docker Image
+      run: |
+        cd ..
+        git clone https://github.com/HERMES-SOC/sdc_aws_processing_lambda.git
+        cd sdc_aws_processing_lambda/lambda_function
+        LINE_TO_INSERT="RUN pip install hermes_core@git+https://github.com/${{ github.event.pull_request.head.repo.full_name }}.git@${{ github.event.pull_request.head.ref }}"
+        sed -i '/RUN pip install -r requirements.txt/a\'"$LINE_TO_INSERT" Dockerfile
+        docker build -t processing_function:latest . --network host
+        
+    - name: Run Lambda Docker Container
+      run: |
+        docker run -d -p 9000:8080 -e USE_INSTRUMENT_TEST_DATA=True processing_function:latest
+        container_id=$(docker ps -qf "ancestor=processing_function:latest")
+        echo "Container ID: $container_id"
+
+    - name: Wait for Container to Initialize
+      run: sleep 5
+
+    - name: Test Lambda Function with curl
+      id: test-lambda
+      run: |
+        cd ../sdc_aws_processing_lambda/
+        # Run curl and write the HTTP status code to a variable
+        HTTP_STATUS=$(curl -XPOST "http://localhost:9000/2015-03-31/functions/function/invocations" -d @lambda_function/tests/test_data/test_eea_event.json)
```

Comment:
Will it be relatively easy to extend this in the future for the other instrument test files?

Reply:
Would it make sense to create a for-each-matrix for something like this?

Reply:
Yeah in the future we can refactor it a bit to be in a matrix, but for now since it's just eea with actual processing code I think it works well.